packages/node/src/typescript.ts

```diff
@@ -491,8 +502,13 @@ export function fixConfig(
   }
 
   // If not specified, the default Node.js module is CommonJS.
+  // However, if package.json has "type": "module" or the file is .mts, use ESNext instead.
   if (config.compilerOptions.module === undefined) {
-    config.compilerOptions.module = 'CommonJS';
+    if (pkg?.type === 'module' || isMtsFile) {
```

Comment:
This will break if the file is `.cjs` and `type: module` which is allowed by node